### PR TITLE
1316 Add GHA to automate weekly label check for issues

### LINKED ISSUE #1316: Evaluate automating the weekly label check
### Overview
Currently the weekly label check is really onerous. You have to go link by link to update add 'missing' labels to all the issues missing any of the required labels.

Requirements:
- On a weekly basis, 
- bot searches size, priority, feature, and role labels
   - creates dynamic filters for each
- bot filters for all issues
   - not containing a size label
      - add https://github.com/hackforla/expunge-assist/labels/size%3A%20missing
   - not containing a priority label
      - add https://github.com/hackforla/expunge-assist/labels/priority%3A%20missing
   - not containing a feature label
      - add https://github.com/hackforla/expunge-assist/labels/feature%3A%20missing 
   - not containing a role label
      - add https://github.com/hackforla/expunge-assist/labels/role%3A%20missing

### Action Items
- [x] evaluate the possibility of automating the weekly label check
   - [ ] if not, message dev lead with explanation why
   - [x] otherwise code GHA to meet the above requirements

### Resources/Notes
reference #369 to see current process

### PR DESCRIPTION
**Pull request by eburdekin**
Fixes #1316 

Quick note - while testing this locally with Act, I actually applied “priority: missing” labels to 4 issues. Glad it worked properly, but sorry about that 😅 I’ll look into why that happened.

**Changes made**: Added a GitHub Action that runs on Sundays and filters through open issues in the repo, adding missing labels where necessary. 

**Reason for changes**: Currently the weekly label check is really onerous. You have to go link by link to update and add `missing` labels to all the issues missing any of the required labels.

I emulated the website team’s GHA setup here for a similar workflow that runs when an issue is opened or assigned:
- Wiki page [here](https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions)
- .yml [here](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/issue-trigger.yml)
- .js [here](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/check-labels.js)

‘What do you think about adding a function to comment on the issues like they have [here](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-issue/add-missing-labels-to-issues/add-labels-template.md)?

### CODE REVIEW

File: .github/workflows/scripts/weekly-label-check.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+// Create Octokit instance
+const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+const REQUIRED_LABELS = ["priority", "role", "feature", "size"];
+const LABEL_MISSING = [
+  "priority: missing",
+  "role: missing",
+  "feature: missing",
+  "size: missing",
+];
```
Comment (sydneywalcoff) [inline .github/workflows/scripts/weekly-label-check.js:12]: lets also add `issue level` as a required label and `issue level: missing` as a missing label. It's not technically part of the audit, but lets encourage people to use them. 


Secondly for scalability, can we create the `LABEL_MISSING` array dynamically from the `REQUIRED_LABELS` array? They all follow the `[REQUIRED_LABELS]: missing` pattern.

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@eburdekin Yay! This is so exciting! I added a comment below with a some small requests, but I'm excited to get this going 🎉

**Comment by eburdekin** [comment]
Thanks @sydneywalcoff ! Should be all set here.

**Review by sydneywalcoff — APPROVED** [review]
@eburdekin yay! Thank you!

**Pull request merged**